bk-nodes/bk_nodes.py

```diff
@@ -0,0 +1,194 @@
+    
+    def post(self):
+        
+        
+        try:
+#request.get
+            postData = request.get_json(force=True, silent=False)
+       
+        except Exception as e:
+            
+            raise ErrorResponse(f"Error parsing json: { sys.exc_info()[0] }  {e}" , status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+        
+        if not postData:
+            raise ErrorResponse(f"Could not parse json." , status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+
+        
+
+        if not isinstance( postData, list ):
+            raise ErrorResponse("array expected", status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+
+        try:
+            bee_db = BeekeeperDB()
+
+            logData = []
+            default_effective_time = datetime.datetime.now(datetime.timezone.utc) # this way all operations in this submission have the exact same time
+            for op in postData:
+                
+                for f in ["node_id", "operation", "field_name", "field_value", "source"]:
+                    if f not in op:
+                        raise Exception(f'Field {f} missing.')
+
+                
+                try:
+                    newLogDataEntry = {
+                        "node_id" : op["node_id"],
+                        "table_name": "nodes_log" , 
+                        "operation": op["operation"],  
+                        "field_name" : op["field_name"], 
+                        "new_value" : op["field_value"] , 
+                        "source": op["source"],
+                        "effective_time" : op.get("effective_time", default_effective_time) }
+
+                except Exception as ex:
+                    raise ErrorResponse(f"Unexpected error: {e}" , status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+
+                logData.append(newLogDataEntry)
+                #print("success", flush=True)
+                
+
+            bee_db.nodes_log_add(logData) #  effective_time=effective_time)
+
+
+
+        
+            
+           
+        except Exception as e:
+            #raise ErrorResponse(f"Unexpected error: { sys.exc_info()[0] } {e}" , status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+            raise ErrorResponse(f"Unexpected error: {e}" , status_code=HTTPStatus.INTERNAL_SERVER_ERROR)
+        
+        response = {"success" : 1}
+        return response
+
+
+    def get(self):
+        return "try POST..."
+
+
+# TODO: add /state/ to get state of all nodes
+# /state/<node_id>
+class State(MethodView):
+    def get(self, node_id):
+
+
+        try:
+
+            bee_db = BeekeeperDB()
+            node_state = bee_db.getNodeState(node_id)
+
+        except Exception as e:
+            raise ErrorResponse(f"Unexpected error: {e}" , status_code=HTTPStatus.INTERNAL_SERVER_ERROR)  
+
+        node_state["timestamp"] = node_state["timestamp"].isoformat()
```

Comment:
if this method is called for a `node_id` that does not exist will this throw an exception since `node_state["timestamp"]` won't exist?

Reply:
Every node state has a timestamp. If node_id does not exist, an exception should have been thrown.

Reply:
oh i thought the call to getNodeState() would just return an empty result.  but either way. your change takes care of this.